src/Config/GatewayAddresses.ts

```diff
@@ -0,0 +1,137 @@
+    this.originAnchorAddress = originAnchorAddress;
+    this.originAnchorOrganizationAddress = originAnchorOrganizationAddress;
+    this.auxiliaryAnchorAddress = auxiliaryAnchorAddress;
+    this.auxiliaryAnchorOrganizationAddress = auxiliaryAnchorOrganizationAddress;
+    this.originGatewayAddress = originGatewayAddress;
+    this.auxiliaryGatewayAddress = auxiliaryGatewayAddress;
+    this.redeemPoolAddress = redeemPoolAddress;
+    this.utilityTokenAddress = utilityTokenAddress;
+    this.originGatewayOrganizationAddress = originGatewayOrganizationAddress;
+    this.auxiliaryGatewayOrganizationAddress = auxiliaryGatewayOrganizationAddress;
+  }
+
+  /**
+   * It provides GatewayAddresses object based from mosaic config object.
+   * @param mosaicConfig Mosaic config object
+   * @param auxChainId Chain id of auxiliary chain.
+   * @returns GatewayAddresses object.
+   */
+  public static fromMosaicConfig(
+    mosaicConfig: MosaicConfig,
+    auxChainId: number,
+  ): GatewayAddresses {
+    if (mosaicConfig.auxiliaryChains[auxChainId]) {
+      return new GatewayAddresses(
+        mosaicConfig.originChain.contractAddresses.valueTokenAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.origin.baseTokenAddress,
+        mosaicConfig.originChain.contractAddresses.stakePoolAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.origin.anchorAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.origin.anchorOrganizationAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.anchorAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.anchorOrganizationAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.origin.eip20GatewayAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.eip20CoGatewayAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.redeemPoolAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.utilityTokenAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.origin.gatewayOrganizationAddress,
+        mosaicConfig.auxiliaryChains[auxChainId].contractAddresses.auxiliary.coGatewayOrganizationAddress,
+      );
+    }
+    throw new Error(`Auxchain id ${auxChainId} is not present in mosaic config`);
+  }
+
+  /**
+   * It provides GatewayAddresses object from gateway config object.
+   * @param gatewayConfig GatewayConfig object.
+   * @returns GatewayAddresses object.
+   */
+  public static fromGatewayConfig(gatewayConfig: GatewayConfig): GatewayAddresses {
+    if(gatewayConfig) {
+      const auxChainId = gatewayConfig.auxChainId;
+      return new GatewayAddresses(
+        gatewayConfig.originContracts.valueTokenAddress,
+        gatewayConfig.originContracts.baseTokenAddress,
+        gatewayConfig.originContracts.stakePoolAddress!,
```

Comment:
Multiple addresses are optional in GatewayConfig. In that case it should be read like 
```
gatewayConfig.stakePoolAddress || gatewayConfig.mosaicConfig.stakePoolAddress
```

Do the same for other addresses also.

Reply:
I have done it for stakepool and redeempool address